### ISSUE TITLE
Executing 2 statements within %%sql fails in DuckDB

### ISSUE THREAD

**Issue by Alex-Monahan** [ploomber/jupysql#674]
Hello! I believe this behavior used to work and that this is a regression. It may be related to the pandas output setting.

```python
%%sql
create temp table my_table as select 42;
select * from my_table;

---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
[<ipython-input-17-8fe54b1c3afd>](https://localhost:8080/#) in <cell line: 1>()
----> 1 get_ipython().run_cell_magic('sql', '', 'create temp table my_table as select 42;\nselect * from my_table;\n')

13 frames
<decorator-gen-124> in execute(self, line, cell, local_ns)

<decorator-gen-123> in execute(self, line, cell, local_ns)

<decorator-gen-122> in execute(self, line, cell, local_ns)

<decorator-gen-121> in execute(self, line, cell, local_ns)

[/usr/local/lib/python3.10/dist-packages/duckdb_engine/__init__.py](https://localhost:8080/#) in __getattr__(self, name)
    104 
    105     def __getattr__(self, name: str) -> Any:
--> 106         return getattr(self.__c, name)
    107 
    108     @property

AttributeError: 'duckdb.DuckDBPyConnection' object has no attribute 'dialect'

```

Here is a google collab that can reproduce the issue: https://colab.research.google.com/drive/1xKjvILJkydM5R095GOVo3hf8446-iZa9?usp=sharing

**Comment by edublancas** [comment]
hi @Alex-Monahan, I was able to reproduce the issue. I believe we introduce this when we added support for generic DB API drivers, we'll fix it!

**Comment by Alex-Monahan** [comment]
Thank you as always! Having this be a living, breathing, improving project is tremendous. Awesome stuff!

**Comment by edublancas** [comment]
hey @Alex-Monahan, update: while working on fixing this we realized that there's some deeper bug involved. so thanks a lot for bringing this up! It'll take a bit more time than expected to fix it but we'll take care of it

**Comment by edublancas** [comment]
hey @Alex-Monahan this has been fixed in 0.9.0, which we released just now! Sorry it took so long, we found out this and other issues were related so we decided to to a big refactoring to close them all at once instead of doing some quick patch.

if you have any other problems, feel free to open another issue!

**Comment by Alex-Monahan** [comment]
Great! It's working again in my test case. Thanks for all the work you to do integrate with DuckDB!
